test/extended/openstack/bug_1765.go

```diff
@@ -0,0 +1,155 @@
+	"github.com/openshift/openstack-test/test/extended/openstack/machines"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/scheme"
+	e2e "k8s.io/kubernetes/test/e2e/framework"
+	e2eskipper "k8s.io/kubernetes/test/e2e/framework/skipper"
+	runtimeclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var _ = g.Describe("[sig-installer][Suite:openshift/openstack] Bugfix", func() {
+	defer g.GinkgoRecover()
+
+	var dc dynamic.Interface
+	var clientSet *kubernetes.Clientset
+	var networkClient *gophercloud.ServiceClient
+
+	g.BeforeEach(func(ctx g.SpecContext) {
+		g.By("preparing openshift dynamic client")
+		cfg, err := e2e.LoadConfig()
+		o.Expect(err).NotTo(o.HaveOccurred())
+		dc, err = dynamic.NewForConfig(cfg)
+		o.Expect(err).NotTo(o.HaveOccurred())
+		clientSet, err = e2e.LoadClientset()
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		skipUnlessMachineAPIOperator(ctx, dc, clientSet.CoreV1().Namespaces())
+
+		g.By("preparing the openstack client")
+		networkClient, err = client.GetServiceClient(ctx, openstack.NewNetworkV2)
+		o.Expect(err).NotTo(o.HaveOccurred(), "Failed to build the OpenStack client")
+	})
+
+	g.Context("bug_1765:", func() {
+		g.It("[Serial] noAllowedAddressPairs on one port should not affect other ports", func(ctx g.SpecContext) {
+			// Check the scenario at https://issues.redhat.com/browse/OCPBUGS-1765
+
+			g.By("Fetching worker machineSets")
+			var rawBytes []byte
+			var newProviderSpec machinev1alpha1.OpenstackProviderSpec
+			var rclient runtimeclient.Client
+
+			machineSets, err := getMachineSets(ctx, dc)
+			o.Expect(err).NotTo(o.HaveOccurred(), "Error getting the workers machinesets")
+
+			if len(machineSets) == 0 {
+				e2eskipper.Skipf("Expects at least one worker machineset. Found none.")
+			}
+
+			cfg, err := e2e.LoadConfig()
+			o.Expect(err).NotTo(o.HaveOccurred(), "Error getting cluster config")
+
+			rclient, err = runtimeclient.New(cfg, runtimeclient.Options{})
+			o.Expect(err).NotTo(o.HaveOccurred(), "Error creating a runtime client")
+
+			// Create an additional network
+			extraNetworkName := fmt.Sprintf("%v-%v", "foonet", RandomSuffix())
+			networkCreateOpts := networks.CreateOpts{Name: extraNetworkName}
+			extraNetwork, err := networks.Create(ctx, networkClient, networkCreateOpts).Extract()
+			o.Expect(err).NotTo(o.HaveOccurred(), "Error creating network")
+			g.By(fmt.Sprintf("Network %v was created", extraNetwork.Name))
+			defer networks.Delete(ctx, networkClient, extraNetwork.ID)
+
+			// Create a machineset and add the extra network to a new machineset
+			err = machinev1.AddToScheme(scheme.Scheme)
+			o.Expect(err).NotTo(o.HaveOccurred(), "Failed to add Machine to scheme")
+			err = configv1.AddToScheme(scheme.Scheme)
+			o.Expect(err).NotTo(o.HaveOccurred(), "Failed to add Config to scheme")
+			newMachinesetParams := framework.BuildMachineSetParams(ctx, rclient, 1)
```

Comment:
Can you please explain how exactly is the machineset Spec defined? Is it reusing the one that exists?